kotlin/viewshed-geoprocessing/src/main/java/com/esri/arcgisruntime/sample/viewshedgeoprocessing/MainActivity.kt

```diff
@@ -0,0 +1,218 @@
+    val fillSymbol = SimpleFillSymbol(SimpleFillSymbol.Style.SOLID, fillColor, null)
+    resultGraphicsOverlay.renderer = SimpleRenderer(fillSymbol)
+
+    val pointSymbol = SimpleMarkerSymbol(
+      SimpleMarkerSymbol.Style.CIRCLE,
+      Color.RED,
+      10F
+    )
+    inputGraphicsOverlay.renderer = SimpleRenderer(pointSymbol)
+
+    mapView.apply {
+      // create a map with the Basemap type topographic
+      map = ArcGISMap(Basemap.Type.TOPOGRAPHIC, 45.3790902612337, 6.84905317262762, 12)
+      // add graphics overlays to the map view
+      graphicsOverlays.addAll(listOf(resultGraphicsOverlay, inputGraphicsOverlay))
+      // add onTouchListener for calculating the new viewshed
+      onTouchListener = object : DefaultMapViewOnTouchListener(
+        applicationContext,
+        mapView
+      ) {
+        override fun onSingleTapConfirmed(e: MotionEvent): Boolean {
+          val screenPoint = android.graphics.Point(
+            e.x.roundToInt(),
+            e.y.roundToInt()
+          )
+          val mapPoint = mMapView.screenToLocation(screenPoint)
+          addGraphicForPoint(mapPoint)
+          calculateViewshedFrom(mapPoint)
+          return super.onSingleTapConfirmed(e)
+        }
+      }
+    }
+  }
+
+  /**
+   * Adds a graphic at the chosen mapPoint.
+   *
+   * @param point in MapView coordinates.
+   */
+  private fun addGraphicForPoint(point: Point) {
+    // remove existing graphics
+    inputGraphicsOverlay.graphics.clear()
+
+    // add new graphic to the graphics overlay
+    inputGraphicsOverlay.graphics.add(Graphic(point))
+  }
+
+  /**
+   * Uses the given point to create a FeatureCollectionTable which is passed to performGeoprocessing.
+   *
+   * @param point in MapView coordinates.
+   */
+  private fun calculateViewshedFrom(point: Point) {
+    // remove previous graphics
+    resultGraphicsOverlay.graphics.clear()
+
+    // cancel any previous job
+    geoprocessingJob?.cancel()
+
+    // create field with same alias as name
+    val field = Field.createString("observer", "", 8)
+
+    // create feature collection table for point geometry
+    val featureCollectionTable =
+      FeatureCollectionTable(listOf(field), GeometryType.POINT, point.spatialReference)
+    featureCollectionTable.loadAsync()
+
+    // create a new feature and assign the geometry
+    val newFeature = featureCollectionTable.createFeature().apply {
+      geometry = point
+    }
+
+    // add newFeature and call perform Geoprocessing on done loading
+    featureCollectionTable.addFeatureAsync(newFeature)
+    featureCollectionTable.addDoneLoadingListener {
+      if (featureCollectionTable.loadStatus == LoadStatus.LOADED) {
+        performGeoprocessing(featureCollectionTable)
+      }
+    }
+  }
+
+  /**
+   * Creates a GeoprocessingJob from the GeoprocessingTask. Displays the resulting viewshed on the map.
+   *
+   * @param featureCollectionTable containing the observation point.
```

Comment:
For param descriptions, its best to separate out the variable name from the description for readability, e.g.
```suggestion
   * @param featureCollectionTable the feature collection table containing the observation point.
```